File: eval/fixtures/algorithms/bfs-bug.js
function solve(graph, start, target) {
  const queue = [[start, 0]], seen = new Set([start]);
  while (queue.length) {
    const [vertex, distance] = queue.pop();
    if (vertex === target) return distance;
    for (const next of graph[vertex]) if (!seen.has(next)) {
      seen.add(next);
      queue.push([next, distance + 1]);
    }
  }
  return -1;
}

File: eval/fixtures/algorithms/bfs-bug.py
def solve(graph, start, target):
    queue = [(start, 0)]
    seen = {start}
    while queue:
        vertex, distance = queue.pop()
        if vertex == target:
            return distance
        for nxt in graph[vertex]:
            if nxt not in seen:
                seen.add(nxt)
                queue.append((nxt, distance + 1))
    return -1
